src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -0,0 +1,149 @@
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ *
+ */
+
+package com.epam.grid.engine.provider.job.slurm;
+
+import com.epam.grid.engine.cmd.GridEngineCommandCompiler;
+import com.epam.grid.engine.cmd.SimpleCmdExecutor;
+import com.epam.grid.engine.entity.CommandResult;
+import com.epam.grid.engine.entity.EngineType;
+import com.epam.grid.engine.entity.JobFilter;
+import com.epam.grid.engine.entity.Listing;
+import com.epam.grid.engine.entity.job.DeleteJobFilter;
+import com.epam.grid.engine.entity.job.DeletedJobInfo;
+import com.epam.grid.engine.entity.job.Job;
+import com.epam.grid.engine.entity.job.JobOptions;
+import com.epam.grid.engine.entity.job.JobLogInfo;
+import com.epam.grid.engine.mapper.job.slurm.SlurmJobMapper;
+import com.epam.grid.engine.provider.job.JobProvider;
+
+import com.epam.grid.engine.provider.utils.CommandsUtils;
+import com.epam.grid.engine.provider.utils.slurm.job.SacctCommandParser;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.thymeleaf.context.Context;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * This class performs various actions with jobs for the SLURM engine.
+ */
+@Service
+@Slf4j
+@ConditionalOnProperty(name = "grid.engine.type", havingValue = "SLURM")
+public class SlurmJobProvider implements JobProvider {
+    private static final String JOB_FILTER = "filter";
+    private static final String SQUEUE_COMMAND = "squeue";
+
+    /**
+     * The MapStruct mapping mechanism used.
+     */
+    private final SlurmJobMapper jobMapper;
+
+    /**
+     * The command execution mechanism used.
+     */
+    private final SimpleCmdExecutor simpleCmdExecutor;
+
+    private final int fieldsCount;
+
+    /**
+     * An object that forms the structure of an executable command according to a template.
+     */
+    private final GridEngineCommandCompiler commandCompiler;
+
+    public SlurmJobProvider(final SlurmJobMapper jobMapper,
+                            final SimpleCmdExecutor simpleCmdExecutor,
+                            final GridEngineCommandCompiler commandCompiler,
+                            @Value("${slurm.job.output-fields-count}") final int fieldsCount) {
```

Comment:
Keep the default value for that constant in `@Value` as well

Reply:
fixed